### PR TITLE
Do not close paragraphs due to div inside svg or mathml.

### LINKED ISSUE #1536: HTML in embedded svg/mathml can auto-close outer p tags
* Mojolicious version: 8.56
* Perl version: 5.32
* Operating system: macOS 10.15

### Steps to reproduce the behavior
Mojo::DOM auto-closes `<p>` tags when it finds an embedded tag like a `<div>`. This is correct. `<math>` or `<svg>` tags can have embedded HTML in them, but it should be treated as entirely isolated from the outside HTML. If a `<p>` tag contains a `<math>` or `<svg>` tag, which has embedded HTML containing a `<div>` tag, Mojo::DOM will auto-close the outer `<p>` tag.

A failing test is shown here: https://github.com/haarg/mojo/commit/07ded055235c09efcd8ec7d1825fdd84210cfdd6

To see a demonstration of how this should parse, see the following:

```
$ curl https://haarg.org/mathml-p-break.html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
  </head>
  <body>
    <div>
      <p>
        <math>
          <semantics>
            <annotation-xml encoding="text/html">
              <div>hi</div>
            </annotation-xml>
          </semantics>
        </math>
      </p>
      <p>
        <div>ho</div>
      </p>
    </div>
  </body>
</html>
$ chrome  --headless --disable-gpu --dump-dom https://haarg.org/mathml-p-break.html 2>/dev/null
<!DOCTYPE html>
<html><head>
    <meta charset="utf-8">
  </head>
  <body>
    <div>
      <p>
        <math>
          <semantics>
            <annotation-xml encoding="text/html">
              <div>hi</div>
            </annotation-xml>
          </semantics>
        </math>
      </p>
      <p>
        </p><div>ho</div>
      <p></p>
    </div>


</body></html>
$ perl -Mojo -e'print g("https://haarg.org/mathml-p-break.html")->dom->to_string'
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
  </head>
  <body>
    <div>
      <p>
        <math>
          <semantics>
            <annotation-xml encoding="text/html">
              </annotation-xml></semantics></math></p><div>hi</div>




      <p>
        </p><div>ho</div>

    </div>
  </body>
</html>
```

### PR DESCRIPTION
**Pull request by mkende**
### Summary
This is a tentative to fix #1536 by not crossing container element while going up the tree to auto-close other tags.

### Motivation
Seemed easy to fix. But then I’m sure that there are plenty of edge cases that I didn’t think of, feel free to suggest tests to add here.

### References
#1536

### CODE REVIEW

File: lib/Mojo/DOM/HTML.pm [hunk 1]
```diff
@@ -182,6 +185,9 @@ sub _end {
     # Ignore useless end tag
     return if $next->[0] eq 'root';
 
+    # Don’t traverse a container tag.
```
Comment (kraih) [inline lib/Mojo/DOM/HTML.pm:188]: Inconsistent comment format.

File: lib/Mojo/DOM/HTML.pm [hunk 2]
```diff
@@ -58,6 +58,9 @@ map { $END{$_} = 'p' } (
   qw(header hgroup hr main menu nav ol p pre section table ul)
 );
 
+# Container HTML elements that create their own scope.
```
Comment (kraih) [inline lib/Mojo/DOM/HTML.pm:61]: Also inconsistent comment format.

**Comment by kraih** [comment]
Yes, it's a complex issue, but the fix seems pretty reasonable.

**Comment by mkende** [comment]
Thanks. I fixed the comment style if you want to take another look.

**Review by kraih — APPROVED** [review]

**Review by marcusramberg — APPROVED** [review]
Looks like a pretty clean fix, with reasonable tests.

**Comment by kraih** [comment]
Thanks, and please squash commits next time.
